packages/flutter/lib/src/widgets/default_text_editing_shortcuts.dart

```diff
@@ -431,4 +427,31 @@ class DefaultTextEditingShortcuts extends Shortcuts {
         return _windowsShortcuts;
     }
   }
+
+  @override
+  Widget build(BuildContext context) {
+    Widget result = child;
+    if (kIsWeb) {
+      // On web platform, this shortcuts makes sure the following:
+      //
+      // 1. Shortcuts fired in TextField are swallowed.
+      // 2. Shortcuts fired above TextField will still trigger corresponding
+      //    intents.
+      //
+      // In (2), the disabling shortcuts will still fire; however, the
+      // [DoNothingAndStopPropagationTextIntent] will not be handled because the
+      // context is above the TextField. The key event will then trigger the
+      // default Text Editing Shortcuts.
+      result = Shortcuts(
```

Comment:
Ah I see, it took me a couple of times reading this code before I even realized there are 2 nested Shortcuts widgets now.

Can you explain these two situations in code?  So in the app below running the code from this PR, I would expect that after tapping the button, ctrl/cmd - A would invoke SelectAllTextIntent, but it would not reach _SelectAllAction in EditableText due to the focus being outside of EditableText?

<details>

<summary>Code</summary>

```dart
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  MyApp({Key? key}) : super(key: key);

  static const String _title = 'Flutter Code Sample';

  final FocusNode _focusNode = FocusNode();

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: _title,
      home: Shortcuts(
        shortcuts: {
          const SingleActivator(LogicalKeyboardKey.keyB, control: true): const SelectAllTextIntent(SelectionChangedCause.keyboard),
          const SingleActivator(LogicalKeyboardKey.keyB, meta: true): const SelectAllTextIntent(SelectionChangedCause.keyboard),
          //const SingleActivator(LogicalKeyboardKey.keyA, control: true): DoNothingIntent(),
          //const SingleActivator(LogicalKeyboardKey.keyA, meta: true): DoNothingIntent(),
        },
        child: Scaffold(
          appBar: AppBar(title: const Text(_title)),
          body: Center(
            child: Column(
              mainAxisAlignment: MainAxisAlignment.center,
              children: <Widget>[
                Focus(
                  focusNode: _focusNode,
                  child: TextButton(
                    onPressed: () {
                      _focusNode.requestFocus();
                    },
                    child: const Text('Press to focus me'),
                  ),
                ),
                const TextField(),
              ],
            ),
          ),
        ),
      ),
    );
  }
}
```

</details>

Reply:
In this situation, in Web ctrl+A will first fire DoNothingAndStopPropagationTextIntent, but no one will handle it since the focus is above the TextField. It will then fire SelectAllTextIntent but still no one will handle it since the focus is above the TextField.

Reply:
Understood and I verified this behavior on web and Mac with my code example 👍